adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -530,6 +598,32 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] {
   }
 
   /**
+   * Prepares two RDDs to be joined with any shuffleRegionJoin. This includes copartition
+   * and sort of the rightRdd if necessary.
+   *
+   *  @param genomicRdd The RDD to join to.
+   *  @param optPartitions An optional number of partitions.
+   *  @return a case class containing all the prepared data for ShuffleRegionJoins
+   */
+  private def prepareForShuffleRegionJoin[X, Y <: GenomicRDD[X, Y], Z <: GenomicRDD[(T, X), Z]](
+    genomicRdd: GenomicRDD[X, Y],
+    optPartitions: Option[Int] = None)(
+      implicit tTag: ClassTag[T], xTag: ClassTag[X]): (RDD[(ReferenceRegion, T)], RDD[(ReferenceRegion, X)]) = {
+
+    val partitions = optPartitions.getOrElse(this.rdd.partitions.length)
+
+    val (leftRdd, rightRdd) = (sorted, genomicRdd.sorted) match {
```

Comment:
Looking elsewhere, `sorted` is assigned as `override val sorted = optPartitionMap.isDefined`. Would it make sense to rename `sorted` &rarr; `hasPartitionMap`?